src/core/interaction-creator.ts

```diff
@@ -0,0 +1,194 @@
+                .getChildrenOfKind(ts.SyntaxKind.JSDocTag)
+                .find((jsDocTag) => jsDocTag.getFirstChildByKind(ts.SyntaxKind.Identifier)?.getText() === 'pact'),
+        );
+        return allPactJsDocsNodes.map(this.getInteractionForPactJsDoc, this);
+    }
+
+    private getInteractionForPactJsDoc(pactJsDoc: tsMorph.Node): Interaction {
+        const nodeWithUsagePactJsDoc = pactJsDoc.getParent()!;
+        const apiFunctionNode = getFunctionNode(nodeWithUsagePactJsDoc);
+        if (apiFunctionNode) {
+            const newInteraction = mapJsDocsIntoInteraction(pactJsDoc);
+            newInteraction.response.status ||= getDefaultResponseStatusForInteraction(newInteraction);
+            newInteraction.description ||= nodeWithUsagePactJsDoc.getFirstChildByKind(ts.SyntaxKind.Identifier)?.getText();
+
+            newInteraction.request.headers = {...this.provider.requestHeaders, ...newInteraction.request.headers};
+            newInteraction.response.headers = {...this.provider.responseHeaders, ...newInteraction.response.headers};
+
+            const responseBody = this.getResponseBodyForApiFunction(apiFunctionNode);
+            const requestBody = this.getRequestBodyForApiFunction(apiFunctionNode);
+            const queryOfRequest = this.getQueryRequestForApiFunction(apiFunctionNode);
+
+            newInteraction.response.body = responseBody.body;
+            newInteraction.response.matchingRules = responseBody.matchingRules;
+            newInteraction.request.body = requestBody?.body;
+            newInteraction.request.query = queryOfRequest?.query;
+            newInteraction.request.matchingRules =
+                queryOfRequest?.matchingRules || requestBody?.matchingRules
+                    ? {...queryOfRequest?.matchingRules, ...requestBody?.matchingRules}
+                    : undefined;
+
+            return newInteraction;
+        }
+        throw Error;
+    }
+
+    private getResponseBodyForApiFunction(apiFunctionNode: tsMorph.Node) {
+        const functionBody = apiFunctionNode.getFirstChildByKindOrThrow(ts.SyntaxKind.Block);
+        let responseBodyType = getResponseTypeFromFunctionBody(functionBody)?.getType();
+        if (responseBodyType === undefined) {
+            responseBodyType = getReturnTypeOfFunction(apiFunctionNode.getType());
+        }
+        const basicTypeRepresentationOfResponse = getBasicRepresentationOfType(responseBodyType, this.sourceFile, true);
+        return {
+            body: changeObjectRepresentationIntoExample(basicTypeRepresentationOfResponse),
+            matchingRules: mapMatchingRepresentation(
+                changeObjectRepresentationIntoMatchingRules(basicTypeRepresentationOfResponse, '$.body'),
+                '$.body',
+            ),
+        };
+    }
+
+    private getRequestBodyForApiFunction(apiFunctionNode: tsMorph.Node) {
+        const requestBodyElement =
+            getParameterWithJsDocFromFunction(apiFunctionNode, 'pact-request-body') ||
```

Comment:
move annotations name into consts or enums

Reply:
Ok,  i've created enum for pact annotations.